index.js

```diff
@@ -19,62 +19,63 @@ const getPowpegRedeemScript = (powpegBtcPublicKeys) => {
     if (!powpegBtcPublicKeys || !(powpegBtcPublicKeys instanceof Array)) {
         throw new Error(ERROR_MESSAGES.INVALID_POWPEG_PUBLIC_KEYS);
     }
+    return getRedeemScriptFromBtcPublicKeys(powpegBtcPublicKeys);
+};
+
+const getRedeemScriptFromBtcPublicKeys = (btcPublicKeys) => {
     // Parse to Buffer and sort keys
-    let defaultPubkeys = powpegBtcPublicKeys
+    const defaultPubkeys = btcPublicKeys
         .map(hex => hex instanceof Buffer ? hex: Buffer.from(hex, 'hex'))
         .sort((a, b) => a.compare(b));
     return bitcoin.payments.p2ms({ m: parseInt(defaultPubkeys.length / 2) + 1, pubkeys: defaultPubkeys }).output;
 };
 
-const getErpRedeemScript = (powpegBtcPublicKeys, erpBtcPublicKeys, csvValue) => {
+const getP2shErpRedeemScript = (powpegBtcPublicKeys, p2shErpBtcPublicKeys, csvValue) => {
     if (!powpegBtcPublicKeys || !(powpegBtcPublicKeys instanceof Array)) {
         throw new Error(ERROR_MESSAGES.INVALID_POWPEG_PUBLIC_KEYS);
     }
-    if (!erpBtcPublicKeys || !(erpBtcPublicKeys instanceof Array)) {
-        throw new Error(ERROR_MESSAGES.INVALID_ERP_PUBLIC_KEYS);
+    if (!p2shErpBtcPublicKeys || !(p2shErpBtcPublicKeys instanceof Array)) {
+        throw new Error(ERROR_MESSAGES.INVALID_P2SH_ERP_PUBLIC_KEYS);
     }
-    if (!csvValue || csvValue.length === 0) {
+    if (!csvValue || isNaN(csvValue)) {
         throw new Error(ERROR_MESSAGES.INVALID_CSV_VALUE);
+    } else {
+        csvValue = numberToHexStringLE(csvValue);
     }
-    
-    let erpPubKeys = erpBtcPublicKeys.map(hex => Buffer.from(hex, 'hex'));
-    let erpRedeemScriptBuff = bitcoin.payments.p2ms({ m: parseInt(erpPubKeys.length / 2) + 1, pubkeys: erpPubKeys });
-    
-    // Remove OP_CHECKMULTISIG from redeem scripts
-    let defaultRedeemScriptWithoutCheckMultisig = getPowpegRedeemScript(powpegBtcPublicKeys).toString('hex').slice(0, -2);
-    let erpRedeemScriptWithoutCheckMultisig = erpRedeemScriptBuff.output.toString('hex').slice(0, -2);
 
-    let bufferLength = parseInt(
+    const defaultRedeemScript = getPowpegRedeemScript(powpegBtcPublicKeys).toString('hex');
+    const p2shErpRedeemScript = getRedeemScriptFromBtcPublicKeys(p2shErpBtcPublicKeys).toString('hex');
```

Comment:
```suggestion
    const emergencyRedeemScript = getRedeemScriptFromBtcPublicKeys(p2shErpBtcPublicKeys).toString('hex');
```
The P2shErp redeem script is the whole script. Its composed of the standard multisig redeem script and the emergency multisig redeem script

Reply:
Resolved